hooks/useIsContract.ts

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from "react";
+
+import { ChainFactory } from "../lib/chainFactory";
+import { EvmClientFactory } from "../lib/evmClient";
+
+const contractCache = new Map<string, boolean>();
+
+export function useIsContract(address: string) {
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (isLoading || contractCache.has(address)) return;
+
+    async function checkContract() {
+      setIsLoading(true);
+      try {
+        const supportedChains = ChainFactory.getSupportedChains();
```

Comment:
Filtering out on testnets/production eliminates half the calls

Reply:
That's already done when invoking `getSupportedChains()`.

https://github.com/hypercerts-org/hypercerts-app/blob/269d801c52b4b7a2534b79a2fb5feae7a65eb773/configs/constants.ts#L32-L34